lib/mock_data/default_data_yaml/default_users_and_roles.yml

```diff
@@ -155,8 +155,8 @@ teachers:
 
   teacher_5:
     login: teacher_with_no_class
-    first_name: Teacher_with_no_class
-    last_name: Teacher_with_no_class
+    first_name: TeacherWithNoClass
```

Comment:
In general it is risky if some characters were valid before, but not anymore.  The problem is that a user with an underscore might go in to modify their password and then when they save the change, Rails will reject it because of their existing name.

I'd suggest trying to add at least the the characters supported by the login fields.
Also if you can come up with a short snippet that would iterate all of the current users in a portal database and print out any that don't match the new pattern.  Something like:
```
User.find_each{|u| puts "#{u.id}: #{u.name}" if !(u.name =~ User.name_regex)
```